### PR TITLE
Add CodePush release-react action

### PR DESCRIPTION
**Pull request by ivan3177**
CodePush is part of AppCenter

This Pull request adds support for deploying updates via CodePush for React Native apps
For usage it also requires installation of AppCenter CLI

All necessary options for `appcenter codepush` CLI command are supported
Updated README with description of action's options and small usage example

### CODE REVIEW

**Comment by goetzc** [comment]
@jp-andre I'm not OP, but there is desire in the community for such Actions using Fastlane, just as when using the other current App Center actions. For example take a look at [this](https://github.com/Pranit-Harekar/fastlane-plugin-codepush#why-this-plugin) third party Action for CodePush or [this](https://github.com/manuelkch/fastlane-plugin-code_push) older one.

**Comment by jp-andre** [comment]
> @jp-andre I'm not OP, but there is desire in the community for such Actions using Fastlane, just as when using the other current App Center actions. For example take a look at [this](https://github.com/Pranit-Harekar/fastlane-plugin-codepush#why-this-plugin) third party Action for CodePush or [this](https://github.com/manuelkch/fastlane-plugin-code_push) older one.

Thanks for your kind answer. I will ping the folks in charge of CodePush and see if they want to have a look at this!

**Comment by andreidubov** [comment]
@jp-andre LGTM

**Comment by alexandergoncharov** [comment]
Hi all,
Thanks for contributing to CodePush!

LGTM. :)

**Comment by tejassharma96** [comment]
Hi @jp-andre, any chance you or someone on the team could take ownership of this pr? It would be really great to have this available in the fastlane action.

File: lib/fastlane/plugin/appcenter/actions/appcenter_codepush_release_react.rb [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+      def self.authors
+        ["Ivan Sokolovskii"]
+      end
+
+      def self.run(params)
+        app = params[:app_name]
+        owner = params[:owner_name]
+        token = params[:api_token]
+        deployment = params[:deployment]
+        dev = paramss[:development]
```
Comment (iv0rish) [inline lib/fastlane/plugin/appcenter/actions/appcenter_codepush_release_react.rb:21]: Fix typo

```suggestion
        dev = params[:development]
```

File: lib/fastlane/plugin/appcenter/actions/appcenter_codepush_release_react.rb [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+                                  optional: true,
+                               description: "Specifies path to where the bundle and sourcemap should be written"),
+          FastlaneCore::ConfigItem.new(key: :sourcemap_output,
+                                      type: String,
+                                  env_name: "APPCENTER_CODEPUSH_SOURCEMAP_OUTPUT",
+                                  optional: true,
+                               description: "Specifies path to write sourcemap to"),
+          FastlaneCore::ConfigItem.new(key: :development,
+                                      type: Boolean,
+                                  env_name: "APPCENTER_CODEPUH_DEVELOPMENT",
```
Comment (iv0rish) [inline lib/fastlane/plugin/appcenter/actions/appcenter_codepush_release_react.rb:144]: Fix typo

```suggestion
                                  env_name: "APPCENTER_CODEPUSH_DEVELOPMENT",
```

File: lib/fastlane/plugin/appcenter/actions/appcenter_codepush_release_react.rb [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+      def self.description
+        "CodePush release react action"
+      end
+
+      def self.authors
+        ["Ivan Sokolovskii"]
+      end
+
+      def self.run(params)
+        app = params[:app_name]
```
Comment (iv0rish) [inline lib/fastlane/plugin/appcenter/actions/appcenter_codepush_release_react.rb:17]: What about change execution directory to parents directory to execute appcenter cli directly?

```suggestion
        Dir.chdir("../".to_s) do
        app = params[:app_name]
```

File: lib/fastlane/plugin/appcenter/actions/appcenter_codepush_release_react.rb [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+          command += "--output-dir #{output} "
+        end
+        if sourcemap_output
+          command += "--sourcemap-output #{sourcemap_output} "
+        end
+        if dry_run
+          UI.message("Dry run!".red + " Would have run: " + command + "\n")
+        else
+          sh(command.to_s)
+        end
```
Comment (iv0rish) [inline lib/fastlane/plugin/appcenter/actions/appcenter_codepush_release_react.rb:61]:
```suggestion
          end
        end
```

**Review by iv0rish — CHANGES_REQUESTED** [review]
Two things have been changed:
 - Move to parents directory before execute appcenter
 - Fix typos

File: lib/fastlane/plugin/appcenter/actions/appcenter_codepush_release_react.rb (hunk 4 above)
Comment (ivan3177) [reply lib/fastlane/plugin/appcenter/actions/appcenter_codepush_release_react.rb:61]: @iv0rish What is going on here? Why add additional indentation? It's an end for `else` clause on line 59.

File: lib/fastlane/plugin/appcenter/actions/appcenter_codepush_release_react.rb (hunk 3 above)
Comment (ivan3177) [reply lib/fastlane/plugin/appcenter/actions/appcenter_codepush_release_react.rb:17]: @iv0rish Thank you for your review! What is motivation for changing directory here?

Comment (iv0rish) [reply lib/fastlane/plugin/appcenter/actions/appcenter_codepush_release_react.rb:17]: Sorry for late response @ivan3177 . Fastlane can't find appcenter cli when I run release-react, so I changed directory where appcenter cli can be executed. You can delete this if it's my own problem.

File: lib/fastlane/plugin/appcenter/actions/appcenter_codepush_release_react.rb (hunk 4 above)
Comment (iv0rish) [reply lib/fastlane/plugin/appcenter/actions/appcenter_codepush_release_react.rb:61]: This is typo. Sorry

**Comment by iv0rish** [comment]
> Hi @iv0rish, anything we could do to get this MR moving?

Hello @sbycrosz, nothing to do for me. Thanks!

**Comment by sbycrosz** [comment]
Hey, 

This PR works as expected on our CI system, so it would be extremely cool to get this merged!

Also, we would like to include this [private-key-path parameter](https://github.com/sbycrosz/fastlane-plugin-appcenter/commit/705b290cab200d82c8968ab64daea94738feee94), What do you think @iv0rish / @ivan3177 

Cheers,
Sam

**Comment by iv0rish** [comment]
> 

LGTM 👍

**Pull request merged**